src/nodes/core/UniformNode.js

```diff
@@ -172,10 +172,10 @@ class UniformNode extends InputNode {
 
 		const sharedNodeType = sharedNode.getInputType( builder );
 
-		const nodeUniform = builder.getUniformFromNode( sharedNode, sharedNodeType, builder.shaderStage, this.name || builder.context.label );
+		const nodeUniform = builder.getUniformFromNode( sharedNode, sharedNodeType, builder.shaderStage, this.name || builder.context.preferredName );
```

Comment:
Maybe `nodeName` instead of `preferredName`? Or are you using `nodeName` already? 👀